SignInCheckIn/SignInCheckIn/Services/RoomService.cs

```diff
@@ -42,11 +42,33 @@ public List<EventRoomDto> GetLocationRoomsByEventId(int eventId)
 
         public EventRoomDto CreateOrUpdateEventRoom(string authenticationToken, EventRoomDto eventRoom)
         {
-            var response = _roomRepository.CreateOrUpdateEventRoom(authenticationToken, Mapper.Map<MpEventRoomDto>(eventRoom));
+            UpdateSubEventRoom(authenticationToken, eventRoom);
+            return UpdateMainEventRoom(authenticationToken, eventRoom);
+        }
 
+        private EventRoomDto UpdateMainEventRoom(string authenticationToken, EventRoomDto eventRoom)
+        {
+            var response = _roomRepository.CreateOrUpdateEventRoom(authenticationToken, Mapper.Map<MpEventRoomDto>(eventRoom));
             return Mapper.Map<EventRoomDto>(response);
         }
 
+        private void UpdateSubEventRoom(string authenticationToken, EventRoomDto eventRoom)
+        {
+            // look to see if there is an AC event for this event and room
+            var acEvents = _eventRepository.GetSubeventsForEvents(new List<int> { eventRoom.EventId }, _applicationConfiguration.AdventureClubEventTypeId);
+            var acEvent = acEvents != null && acEvents.Any() ? acEvents.FirstOrDefault() : null;
+
+            // if there is an ac event see if it has the room
+            if (acEvent == null) return;
+            // if it has the room update the room
+            var acEventRoom = _roomRepository.GetEventRoom(acEvent.EventId, eventRoom.RoomId);
+            if (acEventRoom == null) return;
+            acEventRoom.Capacity = eventRoom.Capacity;
+            acEventRoom.Volunteers = eventRoom.Volunteers;
+            acEventRoom.AllowSignIn = eventRoom.AllowSignIn;
```

Comment:
Are these two objects not the same type of object? If they are, would it not be easier to do acEventRoom = eventRoom?

I don't know this particular feature well enough oviously.